Lib/test/test_py3kwarn.py

```diff
@@ -247,6 +247,13 @@ def test_file_xreadlines(self):
             with check_py3k_warnings() as w:
                 self.assertWarning(f.xreadlines(), w, expected)
 
+    def test_file_open(self):
+        expected = ("The built-in file open function is not supported in 3.x, ",
+                    "use the io.open function instead with the encoding keyword argument")
```

Comment:
Isn't the standard idiom to use `file` instead of `open`? I think `file` works the same in 2.7 and 3.x?

Reply:
Yes, my bad, I left the test to use `file` while testing `open`, unless you also think since its not mainstream, we shouldn't warn.

Reply:
I also mean shouldn't the message say something like "'open' isn't supported in 3.x: use 'file' instead"?

Reply:
I'm confused. `open` is present and works just fine in Python3:
```
arrakis:edd> python3.9
Python 3.9.13 (main, Aug 29 2022, 13:12:10) 
[Clang 13.0.0 ] on openbsd7
Type "help", "copyright", "credits" or "license" for more information.
>>> open("/etc/hosts")     
<_io.TextIOWrapper name='/etc/hosts' mode='r' encoding='UTF-8'>
```

Reply:
The Python 3 implementation is available in io.open where we set an encoding option e.g `with open("/etc/hosts", encoding='utf-8') as f:` The builtin `open` remained in >=2.6 versions but its recommended to use `io.open` with the new syntax.

Reply:
It's just that I don't see any mention of this in the [Python docs](https://docs.python.org/3/library/functions.html#open).

Reply:
I see where Joannah's coming from: if you read https://docs.python.org/3/howto/pyporting.html it says "open is subtly different between Python 2 and 3, but Python 2's open is the same as Python 3's io.open". At least I *think* that's what it says! @nanjekyejoannah ?

Reply:
@vext01  Actually to elaborate the builtin `open()` exists in both Py2.x and Py3.x with different syntax i.e:
In 2.x:

```
>>> open("/etc/hosts")
<open file '/etc/hosts', mode 'r' at 0x10f33ee40>
>>> import io
>>> io.open("/etc/hosts")
<_io.TextIOWrapper name='/etc/hosts' encoding='UTF-8'>
>>> 
```

In 3.x:
```
>>> open("/etc/hosts") 
<_io.TextIOWrapper name='/etc/hosts' mode='r' encoding='UTF-8'>
>>> import io
>>> io.open("/etc/hosts")
<_io.TextIOWrapper name='/etc/hosts' mode='r' encoding='UTF-8'>
>>>  
```

Note, in Py3.x both the built in `open()` and `io.open()` are similar to `io.open()` in Python versions >=2.6.

Reply:
> I also mean shouldn't the message say something like "'open' isn't supported in 3.x: use 'file' instead"?

@ltratt `file` is equivalent to the builtin `open()` in py2.x:

```
>>> open("/etc/hosts")
<open file '/etc/hosts', mode 'r' at 0x10f33ee40>
>>> file("/etc/hosts")
<open file '/etc/hosts', mode 'r' at 0x10f33ee40>
>>> 
```

In py3.x, the same fails:

```
>>> file("/etc/hosts")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'file' is not defined
```

Dont you think the messaging in the fix statement should relate to `open` for this? We can handle `file` separately, no?

Reply:
@nanjekyejoannah I'm saying that I now understand why pygrate2 says use `io.open` (i.e. I think this PR is correct).

Reply:
So the discrepancy we are warning about here is the difference between `_io.TextIOWrapper` and `open file`?

Reply:
>So the discrepancy we are warning about here is the difference between _io.TextIOWrapper and open file?

Isn't `TextIOWrapper` just a return type?

Reply:
Yes, but the type differs between python2 and 3.

Reply:
OK, I think I get it, so disregard the above comment.